docs/features/collision/collision.md

```diff
@@ -5,11 +5,137 @@ title: Collision
 ---
 
-:::warning[Not up to date]
-This page **is not** up to date for MonoGame.Extended `@mgeversion@`.  If you would like to contribute to updating this document, please [create a new PR](https://github.com/craftworkgames/craftworkgames.github.io/pulls)
+:::tip[Up to date]
+This page is **up to date** for MonoGame.Extended `@mgeversion@`.  If you find outdated information, [please open an issue](https://github.com/craftworkgames/craftworkgames.github.io/issues).
 :::
 
-## Usage
+## Collision basics
+
+To use the Collision code, you need to perform the following 4 steps:
+1. Create a class that implements the `ICollisionActor` interface.
+1. Implement the `OnCollision` method in your class.  This defines what happens when something hits your object.
+1. Create a `CollisionComponent` instance defining the bounds of your world.
+1. `Insert` an instance of your class into the CollisionComponent.
+
+The [Full Example](#full-example) uses an extra Interface IEntity that's optional but can be helpful.
+
+## Detailed Explanation
+
+There are a few different items that belong to the collision Namespace.
+1. Space Algorithms
+1. Layer
+1. CollisionComponent
+1. ICollisionActor
+
+You don't need to understand them in detail to use them.  Below are high level explanations of each.
+
+You may skip to the [Full Example](#full-example) section and ignore these details if you wish.
+
+### Space Algorithms
+
+(Advanced/Optional: You don't need to create one of these, this is just explaining how things work behind the scenes.  Skip to the [Full Example](#full-example) for the bare minimum.)
+
+Currently there are 2 Space Algorithms implemented:
+1. `QuadTree`
+1. `SpatialHash`
+
+#### Space Algorithms: QuadTree
+
+A `QuadTree` is a data structure that starts off with a single rectangular area.  Entities are added, and if they reach the maximum number for that rectangular area (25 by default), the area is split up into 4 equal size parts or Quadrants.  This can continue until the maximum depth is reached (7 by default).
+
+The benefit is that you reduce the number of entities you have to check collisions on, since you keep partitioning the screen into smaller and smaller sets of entities.
+
+The management class is `QuadTreeSpace` which uses the `QuadTree`.
+
+Example creation of a QuadTreeSpace:
+```csharp
+QuadTreeSpace quadTreeSpace = new QuadTreeSpace(new RectangleF(x, y, width, height));
+```
+See [QuadTrees](https://en.wikipedia.org/wiki/Quadtree) on Wikipedia for generic more information.
+
+#### Space Algorithms: SpatialHash
+
+Think of mipmaps or approximations.  The screen is split up into N sections, and the object is either in that large section or not.
+
+Example creation of a SpatialHash:
+```csharp
+SpatialHash shash = new SpatialHash(new Vector2(32, 32));
+```
+See [Spatial Hashing](https://www.gamedev.net/tutorials/programming/general-and-gameplay-programming/spatial-hashing-r2697/) on GameDev.
+
+### Layer
```

Comment:
Wasn't sure where to put this review note so I'll do it here at the `Layer` header.  Since `Layers` are "Advanced/Optional", and the full example below doesn't make explicit use of them, it may be better to move the `Layer` section to the bottom of the document.  

For example, the first part of the document is explaining the minimum setup for collisions being the `CollisionComponent` and the `ICollisionActor` followed by the example.  Then after the example is shown, have an `Advanced` section that discusses the `Layer` object, reflect back on how the `CollisionComponent` discussed before uses the `QuadTree` by default and you can change this by creating your own layer.

So basically reorganizing the docs so it starts with the requirement pieces and then has a section on advanced use cases. The space algorithms would be moved down to that part of the documents as well.

I may be explaining this poorly, so feel free to reach out to me and we can talk more.

Reply:
Gotcha, so sort of separate it into beginner/basic and advanced?

Reply:
@AristurtleDev  I'm still having trouble thinking of how to reorganize what I wrote up.

My main concern is that jumping straight into the example would be overwhelming for most, and I want to explain the different parts of the collision classes and how they interact with eachother.

I'm worried that leaving that out would be confusing.

Reply:
I'll take a stab at it and send an update